core/good_key.go

```diff
@@ -38,53 +38,163 @@ var (
 	smallPrimes          []*big.Int
 )
 
+// KeyPolicy etermines which types of key may be used with various boulder
+// operations.
+type KeyPolicy struct {
```

Comment:
Please put this config object in cmd, instead. We're trying hard to keep our config objects out of the production code paths.

Also, it's name is a bit vague. We have lots of keys and policy things. How about "AllowedSigningAlgos"?

Then you can also drop the "allow" from all of these things names.